static/app/components/core/tabs/tab.tsx

```diff
@@ -60,9 +61,10 @@ export interface BaseTabProps {
 function InnerWrap({
   children,
   to,
+  disabled,
   orientation,
-}: Pick<BaseTabProps, 'children' | 'to' | 'orientation'>) {
-  return to ? (
+}: Pick<BaseTabProps, 'children' | 'to' | 'orientation' | 'disabled'>) {
+  return to && !disabled ? (
```

Comment:
this is the core change - disabled tabs with a `to` prop aren’t rendered as `TabLink`, but as a normal disabled `span` element instead.